File: mini-package-gui/src/app/models/paging.request.ts
import { Filter } from "./filter.model";
import { SortFilter } from "./sort.filter";

export class PagingRequest {
  id  : number;
  role: string;
  onlyOffers: boolean;
  numberOfRows: number;
  pageNumber: number;
  filters: Filter[]
  sortFilter: SortFilter | null;

  constructor() {
    this.id = 0;
    this.role = "";
    this.onlyOffers = false;
    this.numberOfRows = 0;
    this.pageNumber = 0;
    this.filters = [];
    this.sortFilter = null;
  }
}

File: mini-package-api/mini-package-api/Models/Requests/Paging/PagingRequest.cs
using mini_package_api.Enums;
using System.Text.Json.Serialization;

namespace mini_package_api.Models.Requests.Paging
{
    public class PagingRequest
    {
        public int Id { get; set; }
        public UserRoleEnum Role { get; set; }
        public bool OnlyOffers { get; set; }
        public int NumberOfRows { get; set; }
        public int PageNumber { get; set; }
        public SortFilter? SortFilter { get; set; }
        public List<Filter> Filters { get; set; }

        [JsonConstructor]
        public PagingRequest()
        {
            
        }
    }
}
